### ISSUE TITLE
Kodi crashes & restarts when accessing JPEGs from a specific camera

### ISSUE THREAD

**Issue by mathieugouin** [mathieugouin/rpiDvr#21]
Go to:
* Top menu
* Pictures
* USB Mounted Ext HD
* Navigate to the large folder

At this point, the list of files are briefly displayed, then kodi crash.  It restarts automatically.

NOTE: Goto http://kodi.wiki/view/Log_file/Easy for reference about uploading the old log.

**Comment by mathieugouin** [comment]
After quick investigation, there seem to have corrupted jpeg in the picture library.

**Comment by mathieugouin** [comment]
The crash can also be reproduced when opening the problematic JPEG from the file manager, therefore not related to thumbnail generation.

**Comment by mathieugouin** [comment]
Check if related to progressive jpg not supported in GPU decoding.

**Comment by mathieugouin** [comment]
https://www.raspberrypi.org/forums/viewtopic.php?f=70&t=92401

**Comment by mathieugouin** [comment]
Currently, affected images are ```DSC*.jpg```

Sony camera.

**Comment by mathieugouin** [comment]
Sony DSC-W510

**Comment by mathieugouin** [comment]
```
Camera make : SONY
Camera model : DSC-W510
Date/Time :2012/05/13 16:51:13
Resolution : 4000 x 3000
Flash used : No
Focal length : 5.6mm (35mm equivalent 31mm)
Exposure time : 0.0020 s (1/500)
Aperture : f/3.0
ISO equiv. : 80
Whitebalance : Auto
Metering Mode : multi spot
Exposure : program (auto)
```

**Comment by mathieugouin** [comment]
[kodi.old.log.txt](https://github.com/mathieugouin/rpiDvr/files/721619/kodi.old.log.txt)

**Comment by mathieugouin** [comment]
https://paste.pound-python.org/show/CgaPIsie0lMepq2XmJjP/

**Comment by mathieugouin** [comment]
Refer to : http://forum.kodi.tv/showthread.php?tid=287441

**Comment by mathieugouin** [comment]
Need to try exiftool to remove metadata?

**Comment by mathieugouin** [comment]
Also http://forum.kodi.tv/showthread.php?tid=251908

**Comment by mathieugouin** [comment]
Check http://exif.regex.info/exif.cgi

**Comment by mathieugouin** [comment]
Working:
![img_6345](https://cloud.githubusercontent.com/assets/11562302/25553064/d0bad346-2c77-11e7-9b83-e59eff8b697d.JPG)

**Comment by mathieugouin** [comment]
Bad camera 1: 
![dsc00166](https://cloud.githubusercontent.com/assets/11562302/25553066/df597dc6-2c77-11e7-9825-f648e5491a6d.JPG)

**Comment by mathieugouin** [comment]
Bad camera 2: 
![p2180004](https://cloud.githubusercontent.com/assets/11562302/25553067/f0e112d4-2c77-11e7-9688-34af2c04b173.jpg)

**Comment by mathieugouin** [comment]
Need to re-submit debug log with repository cleaned-up (clean-up done)

**Comment by mathieugouin** [comment]
Camera: Olympus E-M5MarkII
* https://picload.org/image/rciggpww/p2180004.jpg
* https://picload.org/image/rciggwrr/p2180006.jpg
* https://picload.org/image/rciggwda/p2180003.jpg

**Comment by mathieugouin** [comment]
Camera: Sony DSC-W510
* https://picload.org/image/rolpocll/dsc00166.jpg
* https://picload.org/image/rolpocgr/dsc00167.jpg
* https://picload.org/image/rolpocra/dsc00168.jpg
* https://picload.org/image/rolpocdi/dsc00196.jpg
* https://picload.org/image/rolpocor/dsc00198.jpg

**Comment by sadortun** [comment]
@mathieugouin  Je sais pas de quand sont ses photos, mais tout mes félicitations à toi et ta femme ! !

**Comment by mathieugouin** [comment]
Salut hihihi...

Merci! J'espère que tout va bien de ton côté!

C'est drôle mais la photo de bedaine est pour mon premier et la photo de
bébé c'est mon deuxième :)

Je ne me rappelais plus que tu étais notifié sur mes notes... Hahah...

Peut être que tu pourrais m'aider? Tu connais un analyseur de JPEG?  J'ai
trouvé http://exif.regex.info/exif.cgi

Je veux comprendre ce qu'il y a de différent entre 2 photos. Pas l'image
mais plus le meta data.

À bientôt!
Mathieu

**Comment by mathieugouin** [comment]
Another debug log without banned repository:

http://paste.ubuntu.com/24478887/

**Comment by mathieugouin** [comment]
Found also an Olympus camera that create the same problem with all JPG.  More data to come...

**Comment by mathieugouin** [comment]
Found another Sony camera with the same issue.

**Comment by mathieugouin** [comment]
Solved w/ Kodi 18.5 using libreelec 9.2.0
